core/src/main/java/org/pentaho/di/core/row/value/ValueMetaBase.java

```diff
@@ -5357,7 +5402,8 @@ public void setPreparedStatementValue( DatabaseMeta databaseMeta, PreparedStatem
               setLength( databaseMeta.getMaxTextFieldLength() );
             }
             String string = getString( data );
-            int len = string.length();
+            //int len = string.length(); - changed to support length in char or bytes
+            int len = getLengthInCharOrBytes( string, databaseMeta );
```

Comment:
You can place the ternary operator here. There is no need for the two methods above.